src/sentry/workflow_engine/migrations/0095_add_unique_constraint_to_datasource.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 5.2.1 on 2025-11-05 01:18
+
+from django.db import migrations, models
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = True
+
+    dependencies = [
+        ("sentry", "1002_group_history_prev_history_remove_db_constraint"),
+        ("workflow_engine", "0094_backfill_issue_stream_detector_workflows"),
+    ]
+
+    operations = [
+        migrations.RemoveIndex(
+            model_name="datasource",
+            name="workflow_en_type_66eafc_idx",
+        ),
+        migrations.AddConstraint(
+            model_name="datasource",
+            constraint=models.UniqueConstraint(
+                fields=("type", "source_id"), name="unique_type_source_id"
+            ),
```

Comment:
Let's switch this ordering around here, otherwise we'll be without an index until the unique constraint builds